### PR TITLE
[JAX] Prepare cross flash attention

### PR DESCRIPTION
**Pull request by zlsh80826**
This PR does
- Prepare the arbitrary attention backend I/O (requires forward output in the backward)
- Add bias support for the JAX bridge code

### CODE REVIEW

File: transformer_engine/jax/cpp_extensions.py [hunk 1]
```diff
@@ -2228,45 +2259,53 @@ class CrossFusedAttnBwdPrimitive(BasePrimitive):
...
+        if attn_bias_type == NVTE_Bias_Type.NVTE_NO_BIAS:
+            bias_shape = (0,)
+        else:
+            bias_shape = (*batch_shape[:-1], 1, q_num_head, q_max_seqlen, kv_max_seqlen)
+        bias_dtype = q_dtype
+
         dq_aval = q_aval.update(shape=q_aval.shape, dtype=q_dtype)
         dkv_aval = kv_aval.update(shape=kv_aval.shape, dtype=kv_dtype)
-        return dq_aval, dkv_aval
+        dbias_aval = q_aval.update(shape=bias_shape, dtype=bias_dtype)
```
Comment (cyanguwa) [inline transformer_engine/jax/cpp_extensions.py:2292]: Should this be bias_aval.update?

Comment (zlsh80826) [reply transformer_engine/jax/cpp_extensions.py:2292]: There is no bias_aval during the backward (calculating the dbias doesn't require the bias). So this is a trick and intended, after `.update`, the `dbias_aval` will have the correct shape and type info.

**Comment by cyanguwa** [comment]
@zlsh80826 I'm seeing some failures with post scale bias and arbitrary backend in my CI for PR 497 (pipeline 75568647). I think your CI passed because those tests are skipped, which is because you don't have `POST_SCALE_BIAS` in your `nvte_get_fused_attn_backend` for `flag_arb`, which makes sense. But could you also please test your changes on top of mine see if the bias related tests can still pass? I tried the same configs on the PyTorch side and they seem to be fine. Thanks!

**Review by cyanguwa — APPROVED** [review]
LGTM

**Pull request merged**